### PR TITLE
Implement APIs for URLs

### LINKED ISSUE #96: Create REST APIs
Create REST APIs that can handle

Get shortcode detail
```GET /api/:code```

Get all the links of the group
```GET /api/:groupCode/```

Get grouped shortcode details
```GET /api/:groupCode/:code```

Create shortcodes
```POST /api/```

### PR DESCRIPTION
**Pull request by Manit1**
Fixes #96

### CODE REVIEW

**Comment by jatinkatyal13** [comment]
@Manit1 Back merge the branch and include the AuthToken authentication that you implemented!

File: src/routes/api/url.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      {
+        longUrl: req.body.longUrl,
+        shortCode: req.body.shortCode,
+        private: req.body.private,
+      },
+      req.user,
+    )
+    if (!url) {
+      throw new Error('Error creating shortlink. Try again')
+    }
```
Comment (jatinkatyal13) [inline src/routes/api/url.ts]: I guess you forgot to send a response. 😄

File: src/routes/api/url.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      error: err.message,
+    })
+  }
+})
+
+route.get('/:group/:code', async (req, res) => {
+  try {
+    const group = await findGroupByPrefix(req.params.group)
+    if (!group) {
+      throw new Error('URL Group prefix not found. Wrong URL possibly.')
```
Comment (jatinkatyal13) [inline src/routes/api/url.ts]: Errors should be raised from the controller.

**Comment by Manit1** [comment]
@jatinkatyal13 Done! 🚀 🎸

**Pull request merged**